src/components/Property/RaiseWorkOrder/RaiseWorkOrderFormView.js

```diff
@@ -133,34 +133,45 @@ const RaiseWorkOrderFormView = ({ propertyReference }) => {
     setError(null)
 
     try {
-      const data = await frontEndApiRequest({
-        method: 'get',
-        path: `/api/properties/${propertyReference}`,
-      })
-      const priorities = await frontEndApiRequest({
-        method: 'get',
-        path: `/api/schedule-of-rates/priorities`,
-      })
-      const trades = await frontEndApiRequest({
-        method: 'get',
-        path: `/api/schedule-of-rates/trades?propRef=${propertyReference}`,
-      })
-      const user = await frontEndApiRequest({
-        method: 'get',
-        path: '/api/hub-user',
-      })
+      const [propertyResponse, priorities, trades, user] = await Promise.all([
+        frontEndApiRequest({
+          method: 'get',
+          path: `/api/properties/${propertyReference}`,
+        }),
+        frontEndApiRequest({
+          method: 'get',
+          path: `/api/schedule-of-rates/priorities`,
+        }),
+        frontEndApiRequest({
+          method: 'get',
+          path: `/api/schedule-of-rates/trades?propRef=${propertyReference}`,
+        }),
+        frontEndApiRequest({
+          method: 'get',
+          path: '/api/hub-user',
+        }),
+      ])
+
+      let contactDetails = []
+
+      if (propertyResponse?.tenure?.id != null) {
+        contactDetails = await frontEndApiRequest({
+          method: 'get',
+          path: `/api/contact-details/${propertyResponse?.tenure?.id}`,
+        })
+      }
 
-      setTenure(data.tenure)
-      setProperty(data.property)
+      setTenure(propertyResponse.tenure)
+      setProperty(propertyResponse.property)
       setPriorities(priorities)
       setTrades(trades)
-      setContacts(data.contactDetails)
+      setContacts(contactDetails)
```

Comment:
Given your setting of contactDetails is conditional, contactDetails could be null here. Does setContact account for that, or should this also be wrapped with a not null?

Reply:
The response class used by the API is unchanged, so it should behave identically. 

I have double checked this The API will return an empty array if no contact details are found